### ISSUE TITLE
linter: false positive in react/exhaustive-deps when dep is used in a default parameter

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#11396]
https://discord.com/channels/1079625926024900739/1080712072012238858/1378013793258307614

```
const myDefaultParam1 = "" // some value
const myFunction = useCallback((param1 = defaultParam1, param2) => {
  // ...
}, [defaultParam1])
    // 👆 error. this should be in deps, but oxlint says otherwise
```
